monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_update_agent_shutdown_status.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+
+from common.agent_events import AgentShutdownEvent
+from common.types import SocketAddress
+from monkey_island.cc.agent_event_handlers import update_agent_shutdown_status
+from monkey_island.cc.models import Agent
+from monkey_island.cc.repository import IAgentRepository, StorageError
+
+AGENT_ID = UUID("1d8ce743-a0f4-45c5-96af-91106529d3e2")
+MACHINE_ID = 11
+CC_SERVER = SocketAddress(ip="10.10.10.100", port="5000")
+
+
+def get_agent_object() -> Agent:
+    return Agent(
+        id=AGENT_ID, machine_id=MACHINE_ID, start_time=0, parent_id=None, cc_server=CC_SERVER
+    )
+
+
+TIMESTAMP = 123.321
+AGENT_SHUTDOWN_EVENT = AgentShutdownEvent(source=AGENT_ID, timestamp=TIMESTAMP)
+
+
+@pytest.fixture
+def agent_repository() -> IAgentRepository:
+    agent_repository = MagicMock(spec=IAgentRepository)
+    agent_repository.upsert_agent = MagicMock()
+    agent_repository.get_agent_by_id = MagicMock(return_value=get_agent_object())
+    return agent_repository
+
+
+def test_update_agent_shutdown_status(agent_repository):
+    update_agent_shutdown_status_handler = update_agent_shutdown_status(agent_repository)
+    update_agent_shutdown_status_handler(AGENT_SHUTDOWN_EVENT)
+    assert agent_repository.get_agent_by_id(AGENT_ID).stop_time == datetime.utcfromtimestamp(
+        TIMESTAMP
+    )
```

Comment:
This passes because the handler manipulates the same object that the test checks, not because the repository was properly updated. Add an `InMemoryAgentRepository` and use that instead.